components/Autocomplete.tsx

```diff
@@ -185,13 +185,13 @@ const Autocomplete: FC<AutocompleteProps> = ({
       >
         <PopoverBody w="100%">
           {optionsToDisplay.length > 0 ? (
-            optionsToDisplay.map(option => (
+            optionsToDisplay.map((option, index) => (
               <Box
                 w="100%"
                 key={
                   typeof option.value === 'object'
-                    ? Object.values(option.value).join('-')
-                    : option.value
+                    ? `${Object.values(option.value).join('-')}-${index}`
+                    : `${option.value}-${index}`
```

Comment:
Using `index` in key is not a good idea.
This link from official React docs.
https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/

Reply:
Fixed